libs/utils/src/celix_hash_map.c

```diff
@@ -562,6 +591,33 @@ void celix_longHashMap_clear(celix_long_hash_map_t* map) {
     celix_hashMap_clear(&map->genericMap);
 }
 
+static bool celix_hashMap_equals(const celix_hash_map_t* map1, const celix_hash_map_t* map2) {
+    if (map1 == NULL && map2 == NULL) {
+        return true;
+    }
+    if (map1 == NULL || map2 == NULL) {
+            return false;
+    }
+    if (map1->size != map2->size) {
+        return false;
+    }
+    for (celix_hash_map_entry_t* entry = celix_hashMap_firstEntry(map1); entry != NULL; entry = celix_hashMap_nextEntry(map1, entry)) {
+        void* value = celix_hashMap_get(map2, entry->key.strKey, entry->key.longKey);
+        if (value == NULL || memcmp(&entry->value, value, sizeof(entry->value)) != 0) {
```

Comment:
This assumes `sizeof(void*) >= sizeof(double)`, which is not true on ARM32. `celix_hashMap_get` should return `celix_hash_map_entry_t` instead.